### PR TITLE
add cookie path

### LINKED ISSUE #13846: SSO requires setting cookie path
See #13823 by  @noctwolf 

The instantiated here is the RequestPathBaseCookieBuilder class.
https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authentication/Cookies/src/CookieAuthenticationOptions.cs#L15
Changed Path in the Build method of the RequestPathBaseCookieBuilder class
https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authentication/Core/src/RequestPathBaseCookieBuilder.cs#L21-L35

`@Tratcher `
Ah, I forgot about those custom builders. Yes, this is a scenario that we should discuss and document but changing it in just this code sample isn't going to help. Please file a doc bug so we can make sure the SSO scenario is covered.

@guardrex 

Cookie options are first discussed here ...

https://docs.microsoft.com/aspnet/core/security/authentication/identity-configuration#cookie-settings

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 99307104-c0d7-3e82-f0b2-af61dcfe072a
* Version Independent ID: 315c02cb-54c4-e386-da85-6e4311ca10c2
* Content: [Share authentication cookies among ASP.NET apps](https://docs.microsoft.com/en-us/aspnet/core/security/cookie-sharing?view=aspnetcore-2.2)
* Content Source: [aspnetcore/security/cookie-sharing.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/security/cookie-sharing.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by noctwolf**
Fixes #13846 
@Rick-Anderson Is this do?

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@noctwolf  we need a write up.  This probably need another H3 section - maybe something like

`### Custom authentication builder`
Custom authentication builder must set the `Cookie.Path` for single sign on applications in `Startup.ConfigureServices`:

```csharp
services.AddDataProtection()
    .PersistKeysToFileSystem({PATH TO COMMON KEY RING FOLDER})
    .SetApplicationName("SharedCookieApp");

services.ConfigureApplicationCookie(options => {
    options.Cookie.Name = ".AspNet.SharedCookie";
    options.Cookie.Path = "/";
});
```

 @guardrex  what do you suggest?

**Comment by guardrex** [comment]
That seems fine.

**Comment by Rick-Anderson** [comment]
@guardrex  can you sign off or should I ask `@HaoK`  ?

**Comment by guardrex** [comment]
@HaoK to confirm.

Then, there are a couple of tiny nits to fix on the PR by us.

**Comment by Rick-Anderson** [comment]
@guardrex  go ahead and fix nits.

**Comment by guardrex** [comment]
Recommend quoting the placeholder because the path would be quoted ...

From ...

```csharp
.PersistKeysToFileSystem({PATH TO COMMON KEY RING FOLDER})
```

... to ...

```csharp
.PersistKeysToFileSystem("{PATH TO COMMON KEY RING FOLDER}")
```

**Comment by HaoK** [comment]
Looks fine to me, @Tratcher should take a look too

File: aspnetcore/security/cookie-sharing.md [hunk 1]
```diff
@@ -40,19 +40,34 @@ In `Startup.ConfigureServices`:
...
     .SetApplicationName("SharedCookieApp");
 
 services.ConfigureApplicationCookie(options => {
     options.Cookie.Name = ".AspNet.SharedCookie";
 });
 ```
 
+### Custom authentication builder
+
+Custom authentication builder must set the `Cookie.Path` for single sign on apps. The following code is added to `Startup.ConfigureServices`:
```
Comment (Tratcher) [inline aspnetcore/security/cookie-sharing.md]: This title and description don't make any sense. What is "Custom authentication builder" and why must Path be set?

```
### Cookie paths

Authentication cookies use the HttpRequest.PathBase as their default Cookie.Path. If your cookie needs to be shared across different base paths then Path must be overridden. The above example can be modified as show here:
```

**Comment by guardrex** [comment]
@Rick-Anderson Updated per @Tratcher + applied a heading to each scenario. Seems clean. :eyes:

**Pull request merged**